iterator.go

```diff
@@ -244,6 +245,22 @@ func (item *Item) EstimatedSize() int64 {
 	return int64(vp.Len) // includes key length.
 }
 
+// ValueSize returns the exact size of the value.
+//
+// This can be called to quickly estimate the size of a value without fetching
+// it.
+func (item *Item) ValueSize() int64 {
+	if !item.hasValue() {
+		return 0
+	}
+	if (item.meta & bitValuePointer) == 0 {
+		return int64(len(item.vptr))
+	}
+	var vp valuePointer
+	vp.Decode(item.vptr)
+	return int64(vp.Len) - int64(len(item.key)) - headerBufSize - crc32.Size - 8 // len(uint64) (timestamp)
```

Comment:
Is that `-8` a valid assumption to make? I'm pretty sure that's the timeout but I'm not sure if this is something I can rely on.

Reply:
The 8 is the timestamp associated with the key. When doing `encodeEntry`, we write the key including the timestamp, but when parsing it via Item, the key is stored without the timestamp (and Item.Version tracks the ts). So, this is right. Could use some comments though, I can add that later.